### ISSUE TITLE
SFT loss NAN 求助

### ISSUE THREAD

**Issue by liuzhangjie1713** [jingyaogong/minimind#505]
104M MiniMind pretrain模型在SFT full_sft_512数据集的第二个epoch出现loss为nan的情况，学习率为代码中默认的5e-7，请问可能是什么原因呢？我接下可以采取的解决方案是什么，是调低epoch或者learning rate重新微调，还是我的pretrain模型就有问题？
训练日志如下

<img width="930" height="494" alt="Image" src="https://github.com/user-attachments/assets/748adce9-1054-4b05-9b41-14f86ce33b72" />
loss图如下：

<img width="1070" height="555" alt="Image" src="https://github.com/user-attachments/assets/4f5b71fb-c45a-4c88-99e4-5bf5b9b47735" />

pretrain 模型是自己在pretrain_hq上预训练的，跑了4epoch，loss图如下

<img width="1027" height="497" alt="Image" src="https://github.com/user-attachments/assets/32bd26f0-fa94-4974-9c1b-68fcefa43a56" />

测试效果如下

<img width="1280" height="674" alt="Image" src="https://github.com/user-attachments/assets/80e82d26-ffe6-44e6-b05e-9930c47327b3" />

机器配置 ：2张4090 ，采用deepseed框架，执行命令为` nohup deepspeed --master_port 29500 --num_gpus=2 train_full_sft.py --use_wandb --epochs=2 --hidden_size=768 --num_hidden_layers=16 --max_seq_len=512 --data_path=/root/autodl-tmp/dataset/sft_512.jsonl > sft_512.log 2>&1 &`

**Comment by jingyaogong** [comment]
还没真正集成DeepSpeed API

换成ddp：
```bash
nohup torchrun --nproc_per_node=2 train_full_sft.py --use_wandb --epochs=2 --hidden_size=768 --num_hidden_layers=16 --max_seq_len=512 --data_path=/root/autodl-tmp/dataset/sft_512.jsonl --learning_rate=1e-7 --grad_clip=0.5 > sft_512.log 2>&1 &
```

然后观察问题

**Comment by jingyaogong** [comment]
@liuzhangjie1713 可以继续补充是否解决，issue先关闭了

**Comment by liuzhangjie1713** [comment]
已按照该方案解决

> 还没真正集成DeepSpeed API
> 
> 换成ddp：
> 
> nohup torchrun --nproc_per_node=2 train_full_sft.py --use_wandb --epochs=2 --hidden_size=768 --num_hidden_layers=16 --max_seq_len=512 --data_path=/root/autodl-tmp/dataset/sft_512.jsonl --learning_rate=1e-7 --grad_clip=0.5 > sft_512.log 2>&1 &
> 然后观察问题
